controllers/job_test.go

```diff
@@ -0,0 +1,126 @@
+package controllers_test
+
+import (
+	"context"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/tinkerbell/rufio/api/v1alpha1"
+	"github.com/tinkerbell/rufio/controllers"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+func TestJobReconcile(t *testing.T) {
+	tests := map[string]struct {
+		machine   *v1alpha1.Machine
+		secret    *corev1.Secret
+		job       *v1alpha1.Job
+		shouldErr bool
+		testAll   bool
+	}{
+		"success taskless job":    {machine: createMachine(), secret: createSecret(), job: createJob("test", createMachine())},
+		"failure unknown machine": {machine: &v1alpha1.Machine{}, secret: createSecret(), job: createJob("test", createMachine()), shouldErr: true},
+		"success power on job":    {machine: createMachine(), secret: createSecret(), job: createJob("test", createMachine(), getAction("PowerOn")), testAll: true},
```

Comment:
Mind splitting the object construction over lines so we don't run off the page?

Reply:
sure thing.